src/server/main.cc

```diff
@@ -11,14 +11,40 @@
 #include "src/proto/protocached.pb.h"
 #include "src/proto/protocached.grpc.pb.h"
 
-void RunServer() {
+// TODO(joejoevictor): Move this to separate header and implementation file
+class ProtocachedServiceImpl final : public protocached::Protocached::Service {
+  grpc::Status Get(
+    grpc::ServerContext* context,
+    const ::protocached::GetRequest* request,
+    protocached::GetResponse* response
+  ) override {
+    protocached::CachedValue cachedValue;
```

Comment:
the cachedValue var is never allocated memory to it... don't we need to create a new instance of this class? If not, wouldn't the set_value down there just arbitrarily overwrite some random place in the memory?

I suck at memory management for C++, especially all the new stuff..

Reply:
I thought when you declare a variable in C++ you already have the memory allocated? It's not initialized but it's allocated. At least that's what I thought how it works so you're right, it's some garbage in it but the question is, do we care? Since there's only 1 field "value" in CachedValue and we're setting it in the next statement anyway, I don't think we need to initialize other stuff in cachedValue? I don't think I know enough about protobuf and it's nuance in c++ to answer this question.